File: NathaliaCardona_GenerarTiempos.py
import numpy as np
import matplotlib.pyplot
import time
#Funcion que retorna el n-esimo numero de la serie de fibonacci
def fibonacci(N):
    #Casos base para 0 y 1
    if N==0:
        return 0
    if(N==1):
        return 1
    #Caso recursivo
    else:
        #segun la formula de fibonacci se calcula recursivamente
        return fibonacci(N-1)+fibonacci(N-2)
#retorna el tiempo que le toma calcular el numero n-esimo de fibonacci
def get_time(N):
    #inicia el cronometro
    t0 = time.time()
    # SOME CODE THAT TAKES TIME
    #corre fibonacci para N
    fibonacci(N)
    #termina de calcular n-esimo termino y para el cronometro
    t1 = time.time()-t0
    return t1

N=35
for i in range(N+1):
    print i,get_time(i)

File: NathaliaCardona_GenerarTiempos.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>
using namespace std;
//Se declaran las funciones y se especifica el tipo de funcion
int fibonacci(int N);
double get_time(int N);
//se crea la clase main
int main(){
  int N=35;
  for(int i=0;i<=N;i++){
    //retorna el N y el tiempo que le tomo calcular ese termino
    cout << i << " " << get_time(i) << endl;
  }

  return 0;
}
//Calcula el numero N de la serie de fibonacci y lo retorna
int fibonacci(int N){
  //se evaluan los dos casos base
  if(N=0){
    return 0;
  }
  if(N=1){
    return 1;
  }
  else{
    return fibonacci(N-1)+fibonacci(N-2);
  }
}
//calcula la cantidad de segundos que le toma calcular el termino n de fibonacci
double get_time(int N){
  //se inicia el cronometro
  clock_t t;
  t = clock();
  // SOME CODE THAT TAKES TIME
  //corre la funcion y cuando termina lo para y retorna el tiempo
  fibonacci(N);
  t = clock() - t;
  float time = ((float)t)/CLOCKS_PER_SEC;
  return time;
}
